### PR TITLE
add more build states

### PR DESCRIPTION
**Pull request by syphar**
This is picking up from the [work by @Nemo157](https://github.com/rust-lang/docs.rs/compare/master...Nemo157:docs.rs:more-build-states), working towards solving #1011 ( and a bunch of other related issues). 

I went through all places I found where we access a `.build_status`, and I believe we need a new "aggregated" build status for a release, which would be: 
- "successful" if there is _any_ successful build, because then we have docs to show, and newer failed or in-progress builds don't change that. 
- then, "failed" when there is any failed build
- the rest is in progress 

In my mind, most of the site doesn't care about in-progress builds, and only cares about failed builds when there is no successful one. We can later opt-in into showing more when we want to. 

I've put the logic into a database view, so now we can use it not only when fetching single releases, but also when we fetch release lists. 
It would be awesome for the reviewer to do the same check for the usages and if my assumption is correct. 

Currently we don't have in-progress builds anywhere, but when we have, we'll have to revisit each handler/view anyways and decide _if_ and _how_ we show in-progress builds or releases. 

The next PR then would be to start adding in-progress crates, releases & builds and then checking everything again.

### CODE REVIEW

**Comment by syphar** [comment]
@Nemo157 I think this is ready for  a first review. 

I tried to take your commits and adapt them to sqlx first, 

and then started to add commits onto these. 

For a review commit-by-commit is probably best, comparing to your commits too.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Changes look good to me, but as usual, please wait for someone else's review before merging.

**Review by Nemo157 — APPROVED** [review]

**Pull request merged**
